zboxcore/sdk/downloader.go

```diff
@@ -37,7 +37,7 @@ type DownloadOptions struct {
 }
 
 // CreateDownloader create a downloander
-func CreateDownloader(allocationID, localPath, remotePath string, opts ...DownloadOption) (Downloader, error) {
+func CreateDownloader(storageSdk *storageSdkSchema, allocationID, localPath, remotePath string, opts ...DownloadOption) (Downloader, error) {
```

Comment:
I don't think it's a good way to pass sdk instance here. Try to init inside function

Reply:
do we need to always init the sdk inside the function? or we can just use the global variable like in the above `var storageSdk = NewStorageSdkSchema()`